pallets/roles/src/lib.rs

```diff
@@ -272,9 +282,25 @@ pub mod pallet {
 				Error::<T>::NotValidator
 			);
 
+			// Get Role key of validator.
+			let validator_id =
+				<T as pallet_session::Config>::ValidatorIdOf::convert(stash_account.clone())
+					.ok_or(Error::<T>::NotValidator)?;
+
+			let session_keys = pallet_session::NextKeys::<T>::get(validator_id)
+				.ok_or(Error::<T>::SessionKeysNotProvided)?;
+			let role_key = OpaqueKeys::get_raw(&session_keys, ROLE_KEY_TYPE);
+
```

Comment:
One problem here is if the validator [rotates](https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot#generating-the-session-keys) their key or removes and changes it, then this old key will no longer work and all their signed submissions will fail. So we need to think of how to handle this, either automated or the validator should manually call another extrinsic in this pallet, maybe another ticket for this?

Reply:
Agreed. Created separate issue to track this problem #408